### DISCUSSION TITLE
3D Label

### DISCUSSION THREAD

**Discussion by InkRobert** [godotengine/godot-proposals#4345] [3D]
Adding text to a 3D scene is quite a nuisance, because you need to create a Viewport in which you need to create a Label, and then render the image in Sprite3D.
**This is not very convenient and creates some inconvenience, and I think that the ability to add text to the scene through a regular 3D Label is a convenient thing that ordinary users need**

![label3d](https://user-images.githubusercontent.com/91805483/161618690-987dd173-6cd0-4a22-b0ec-d6338b6f4426.png)

**_I know that there is at least one addon that implements this, but this again - this should be in Godot Engine initially, because not everyone wants to look for the very addon that can fully implement the functionality you need_**

**Other game engines have had such an elementary thing for a long time, so why should we look for workarounds to the implementation?**

**Comment by Mickeon** [comment]
Yeah, it seems like something that is generally agreed upon. https://github.com/godotengine/godot-proposals/issues/3219

**Reply by Calinou** [reply]
For reference, a Label3D node is now being worked on: https://github.com/godotengine/godot/pull/60386

**Reply by SimplyPhy** [reply]
I believe this is now complete and merged into 4.0 Alpha 7.

**Comment by Calinou** [comment]
> Adding text to a 3D scene is quite a nuisance, because you need to create a Viewport in which you need to create a Label, and then render the image in Sprite3D.

Remember that using a projected Label node performs better and is often more readable in practice. See the [3D waypoints demo](https://github.com/godotengine/godot-demo-projects/tree/master/3d/waypoints) for an example.

**Reply by Zireael07** [reply]
Even more reason to have a built-in way to do this instead of people stumbling around in search of the best solution

**Reply by Yagich** [reply]
The problem with projected UI elements is that they are not really in a 3D space, so it won't change the size depending on the distance to the camera and be always facing the camera like a billboard, which is not always desired.
